### ISSUE TITLE
the loss function

### ISSUE THREAD

**Issue by lhr-30** [jingyang2017/emonet_train#1]
I'm reproducing this paper now and I have some trouble training this net. I found that I wrote the same loss function to yours, but the result seems that the prediction of the heatmap did not get trained. But in your code there is also no loss function about the heatmap and FAN part. So I wonder how it works? thanks

**Comment by jingyang2017** [comment]
From my understanding, the feature maps from FAN is seen as the input to a learnable network. FAN part is pretrained on face alignment datasets and is fixed in emotion recognition task. FAN functions as a feature provider.

**Comment by jingyang2017** [comment]
> I'm reproducing this paper now and I have some trouble training this net. I found that I wrote the same loss function to yours, but the result seems that the prediction of the heatmap did not get trained. But in your code there is also no loss function about the heatmap and FAN part. So I wonder how it works? thanks


From my understanding, the feature maps from FAN is seen as the input to a learnable network. FAN part is pretrained on face alignment datasets and is fixed in emotion recognition task. FAN functions as a feature provider.

**Comment by lhr-30** [comment]
I mean how do you train the FAN net? or do you mean we already have a FAN net that is well-trained? if so, how can I get the well-trained hyper-parameter of the FAN.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年8月9日(星期二) 下午4:32
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [jingyang2017/emonet_train] the loss function (Issue #1)





  
I'm reproducing this paper now and I have some trouble training this net. I found that I wrote the same loss function to yours, but the result seems that the prediction of the heatmap did not get trained. But in your code there is also no loss function about the heatmap and FAN part. So I wonder how it works? thanks
  
From my understanding, the feature maps from FAN is seen as the input to a learnable network. FAN part is pretrained on face alignment datasets and is fixed in emotion recognition task. FAN functions as a feature provider.

**Comment by lhr-30** [comment]
By the way, I send a message to your CSDN account(sorry to bother you but I think it may be more convenient).So I really hope that I can get a reply. Thanks a lot!


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "刘鸿睿"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年8月9日(星期二) 晚上6:06
***@***.******@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;回复： [jingyang2017/emonet_train] the loss function (Issue #1)



 I mean how do you train the FAN net? or do you mean we already have a FAN net that is well-trained? if so, how can I get the well-trained hyper-parameter of the FAN.




------------------ 原始邮件 ------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年8月9日(星期二) 下午4:32
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [jingyang2017/emonet_train] the loss function (Issue #1)





  
I'm reproducing this paper now and I have some trouble training this net. I found that I wrote the same loss function to yours, but the result seems that the prediction of the heatmap did not get trained. But in your code there is also no loss function about the heatmap and FAN part. So I wonder how it works? thanks
  
From my understanding, the feature maps from FAN is seen as the input to a learnable network. FAN part is pretrained on face alignment datasets and is fixed in emotion recognition task. FAN functions as a feature provider.

**Comment by lhr-30** [comment]
Can you tell me where you get the pretrained hyperparameter of the FAN? I tried to find it but didn't get it?

**Comment by jingyang2017** [comment]
> Can you tell me where you get the pretrained hyperparameter of the FAN? I tried to find it but didn't get it?

You can extracted the FAN pretrained parameters from https://github.com/face-analysis/emonet/tree/master/pretrained. Or you can use https://github.com/hhj1897/fan_training/tree/master/pretrained

**Comment by lhr-30** [comment]
Thanks for your help, now I can run the FAN part correctly. But I still get a result not good as the paper said. I trained my network on th AffectNet dataset with 26w pictures to train the 8-classes net.I only got 60% ACC and PCC、CCC results were also not that good. Can  you tell me your results? Are there any other data augmentation methods used?

**Comment by jingyang2017** [comment]
> Thanks for your help, now I can run the FAN part correctly. But I still get a result not good as the paper said. I trained my network on th AffectNet dataset with 26w pictures to train the 8-classes net.I only got 60% ACC and PCC、CCC results were also not that good. Can you tell me your results? Are there any other data augmentation methods used?

I have put my trained model in https://github.com/jingyang2017/emonet_test/tree/master/ibug/emotion_recognition/emonet/weights. I get very close results as the paper said.
